timezone/src/modules/Timezone.rb

```diff
@@ -472,53 +472,72 @@ def CallMkinitrd
 
     # Set the new time and date given by user
     def SetTime(year, month, day, hour, minute, second)
-      if !Arch.s390
-        date = Builtins.sformat(
-          " --date=\"%1/%2/%3 %4:%5:%6\" ",
-          month,
-          day,
-          year,
-          hour,
-          minute,
-          second
-        )
-        cmd = ""
-        if Ops.greater_than(Builtins.size(@timezone), 0) &&
-            @hwclock != "--localtime"
-          cmd = Ops.add(Ops.add("TZ=", @timezone), " ")
-        end
-        cmd = Ops.add(
-          Ops.add(Ops.add(cmd, "/sbin/hwclock --set "), @hwclock),
-          date
-        )
-        Builtins.y2milestone("SetTime cmd %1", cmd)
-        SCR.Execute(path(".target.bash"), cmd)
-        cmd = Ops.add("/sbin/hwclock --hctosys ", @hwclock)
-        Builtins.y2milestone("SetTime cmd %1", cmd)
-        SCR.Execute(path(".target.bash"), cmd)
-        # actually, it was probably not called, but do not let it change the time again after manual change
-        @systz_called = true
-      end
+      return nil if Arch.s390
+
+      timedate = "#{month}/#{day}/#{year} #{hour}:#{minute}:#{second}"
 
+      if set_hwclock(timedate)
+        sync_hwclock_to_system_time
+      else
+        # No hardware clock available (bsc#1103744)
+        log.info("Fallback: Leaving HW clock untouched, setting only system time")
+        set_system_time(timedate)
+      end
       nil
     end
 
     # Set the Hardware Clock to the current System Time.
     def SystemTime2HWClock
-      if !Arch.s390
-        cmd = ""
-        if Ops.greater_than(Builtins.size(@timezone), 0) &&
-            @hwclock != "--localtime"
-          cmd = Ops.add(Ops.add("TZ=", @timezone), " ")
-        end
-        cmd = Ops.add("/sbin/hwclock --systohc ", @hwclock)
-        Builtins.y2milestone("cmd %1", cmd)
-        SCR.Execute(path(".target.bash"), cmd)
-      end
+      return nil if Arch.s390
 
+      cmd = tz_prefix + "/sbin/hwclock --systohc #{@hwclock}"
+      log.info("cmd #{cmd}")
+      SCR.Execute(path(".target.bash"), cmd)
       nil
     end
 
+    # Set the hardware clock with the given date.
+    # @param timedate [String]
+    # @return [Bool] true if success, false if error
+    #
+    def set_hwclock(date)
+      cmd = tz_prefix + "/sbin/hwclock --set #{@hwclock} --date=\"#{date}\""
+      log.info("set_hwclock: #{cmd}")
+      SCR.Execute(path(".target.bash_output"), cmd) == 0
```

Comment:
The `.target.bash_output` call returns a `Hash`, looks like the `.target.bash` should be used here instead...

Reply:
You are right.

For a while I had considered checking the error message of that command for that specific text, but then I decided against it and to just always try to set the system time when `hwclock` fails to set it via the RTC.